### PR TITLE
JS Style Guide v1

### PR DESCRIPTION
**Pull request by epetrsn**
Initial attempt at defining the ProsperWorks JavaScript Style Guide. These rules should represent the current state of the eslint config located at [ALI/client/tools/.eslintrc](https://github.com/ProsperWorks/ALI/blob/master/client/tools/.eslintrc) and as explained in the old [wiki](https://sites.google.com/a/mobiusmediagroup.com/eli/engineering/client/playbook/javascript-guidelines).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -914,71 +901,50 @@ Other Style Guides
...
+    };
+
+    // bad
     [1, 2, 3].map(number => {
       const nextNumber = number + 1;
       `A string containing the ${nextNumber}.`;
     });
 
-    // good
+    // bad
```
Comment (apellerano-pw) [inline README.md:924]: 😆

**Review by apellerano-pw — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (epetrsn) [reply README.md:924]: It's so easy!

File: README.md [hunk 2]
```diff
@@ -1,17 +1,11 @@
-# Airbnb JavaScript Style Guide() {
+# PropserWorks JavaScript Style Guide() {
```
Comment (ghost) [inline README.md]: s/Propser/Prosper/ :P

File: README.md [hunk 3]
```diff
@@ -376,9 +366,6 @@ Other Style Guides
       return x * y;
     });
 
-    // good
-    [1, 2, 3].map(x => x + 1);
-
```
Comment (ghost) [inline README.md:381]: booooooo (it's cool i'll live with it)

File: README.md [hunk 4]
```diff
@@ -1915,20 +1861,24 @@ Other Style Guides
...
     // good
-    if (test) return false;
+    if (test) { return false; }
 
     // good
     if (test) {
       return false;
     }
 
-    // bad
```
Comment (ghost) [inline README.md:1931]: yea i also don't understand why this is bad; this is useful inside of CPs for defining simple `get` and `set` functions.

**Review by ghost — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (epetrsn) [reply README.md]: Good catch!

File: README.md (hunk 4 above)
Comment (epetrsn) [reply README.md:1931]: I think if we rewrite this rule to say "always use braces, even with single-line blocks", this should become "good".

File: README.md [hunk 5]
```diff
@@ -3201,129 +3151,10 @@ Other Style Guides
...
-  - **Syzygy Warsaw**: [syzygypl/javascript](https://github.com/syzygypl/javascript)
-  - **Target**: [target/javascript](https://github.com/target/javascript)
-  - **TheLadders**: [TheLadders/javascript](https://github.com/TheLadders/javascript)
-  - **The Nerdery**: [thenerdery/javascript-standards](https://github.com/thenerdery/javascript-standards)
-  - **T4R Technology**: [T4R-Technology/javascript](https://github.com/T4R-Technology/javascript)
-  - **VoxFeed**: [VoxFeed/javascript-style-guide](https://github.com/VoxFeed/javascript-style-guide)
-  - **WeBox Studio**: [weboxstudio/javascript](https://github.com/weboxstudio/javascript)
-  - **Weggo**: [Weggo/javascript](https://github.com/Weggo/javascript)
-  - **Zillow**: [zillow/javascript](https://github.com/zillow/javascript)
-  - **ZocDoc**: [ZocDoc/javascript](https://github.com/ZocDoc/javascript)
```
Comment (golivera) [inline README.md:3289]: should we request to be added to this list 😆 ?

File: README.md (hunk 4 above)
Comment (ghost) [reply README.md:1931]: ah okay, sounds good to me

**Pull request merged**